src/messenger/messenger.cpp

```diff
@@ -461,12 +461,17 @@ void Messenger::SendMessage(common::ManagedPointer<ConnectionRouter> router_id,
 
   // Build and send the message. Note that ConnectionRouter is a ROUTER socket.
   zmq::message_t router_data(recv_id.data(), recv_id.size());
-  if (router_id->socket_->send(router_data, zmq::send_flags::sndmore).has_value()) {
-    ZmqMessage reply = ZmqMessage::Build(send_msg_id, recv_cb_id, router_id->identity_, message);
-    ZmqUtil::SendMsgIdentity(common::ManagedPointer(router_id->socket_.get()), router_id->identity_);
-    ZmqUtil::SendMsgPayload(common::ManagedPointer(router_id->socket_.get()), reply);
-    MESSENGER_LOG_TRACE(fmt::format("[PID={}] Messenger ({}) SENT-TO {}: {} ", ::getpid(), router_id->identity_,
-                                    recv_id, reply.GetRawPayload()));
+  try {
```

Comment:
This is the first thing where I think we should discuss briefly. I think it makes more sense for the **caller** of this function to catch the exception so that they have a choice. If we need to expose some debugging information I'm OK with putting all relevant debug information into the exception, see the exceptions that currently take error codes.

Reply:
I see, having the caller handle any exceptions makes sense. I will revert the change.